### PR TITLE
[Crypto] Implement OS::get_entropy and CryptoCore::RandomGenerator

### PR DESCRIPTION
**Pull request by Faless**
In this PR:

- Add get_entropy to OS
  Implemented via `BCryptGenRandom` on Windows.
  Implemented via `getentropy` syscall when available.
  Implemented via `/dev/urandom` device as a fallback.
  The `/dev/urandom` fallback can be disabled via the `NO_URANDOM` build flag.
  Note: The HTML5 version relies on emscripten file system urandom device which itself uses the Crypto API when available or the plain old not crypto-safe `Math.random()` otherwise.
- Implement CryptoCore::RandomGenerator.
  As a cryptographically secure random generator.
  Internally it uses mbedTLS CTR-DRBG implementation which gets re-seeded with entropy from OS::get_entropy when needed.
  CryptoCore now additionally depends on `ctr_drbg.c` and `entropy.c` thirdparty mbedtls files.
- ResourceUID now uses CryptoCore::RandomGenerator for generating IDs.
- OSX Codesign exporter now uses CryptoCore RNG.

Closes #55181 (superseded).

### CODE REVIEW

**Review by bruvzg — COMMENTED** [review]
> OSX Codesign exporter now uses CryptoCore RNG.

I do not think it's really necessary for an ad-hoc signature, but seems OK.

I have also tested both `getentropy` and `/dev/urandom` branches on macOS, and both work fine.

If I remember correctly, `getentory` is a private API on iOS, and can't be used, but `/dev/urandom` should work.

**Comment by reduz** [comment]
Amazing work!

File: core/io/resource_uid.cpp [hunk 1]
```diff
@@ -31,9 +31,10 @@
 #include "resource_uid.h"
 
 #include "core/config/project_settings.h"
-#include "core/crypto/crypto.h"
+#include "core/crypto/crypto_core.h"
 #include "core/io/dir_access.h"
 #include "core/io/file_access.h"
+#include "core/os/os.h"
```
Comment (akien-mga) [inline core/io/resource_uid.cpp:37]: I don't see what uses `OS` right away, is it actually needed, or is there a more specific header we can include instead?

File: drivers/unix/os_unix.cpp [hunk 2]
```diff
@@ -150,6 +165,31 @@ String OS_Unix::get_stdin_string(bool p_block) {
...
+Error OS_Unix::get_entropy(uint8_t *r_buffer, int p_bytes) {
+#if defined(UNIX_GET_ENTROPY)
+	int left = p_bytes;
+	int ofs = 0;
+	do {
+		int chunk = MIN(left, 256);
+		ERR_FAIL_COND_V(getentropy(r_buffer + ofs, chunk), FAILED);
+		left -= chunk;
+		ofs += chunk;
+	} while (left);
```
Comment (akien-mga) [inline drivers/unix/os_unix.cpp:177]: A bit cleaner if explicitly treated as int and not bool (same below):
```suggestion
	} while (left > 0);
```

**Review by akien-mga — APPROVED** [review]

**Comment by Faless** [comment]
> I do not think it's really necessary for an ad-hoc signature, but seems OK.

Yeah, I agree this is a bit of an overkill, I used that because I wanted to avoid using `get_entropy` directly (due to potential starvation on older systems).
Even if in that is not a problem in the code signer specifically, I would prefer to avoid having potential "bad examples" that could mislead other contributors.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**